pkg/frontend/adminactions/resources_list_scoped.go

```diff
@@ -0,0 +1,48 @@
+package adminactions
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	mgmtcompute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2020-06-01/compute"
+
+	"github.com/Azure/ARO-RP/pkg/util/arm"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+func (a *azureActions) ResourcesListScoped(ctx context.Context, resourceType string) ([]byte, error) {
+	clusterRGName := stringutils.LastTokenByte(a.oc.Properties.ClusterProfile.ResourceGroupID, '/')
+
+	filter := fmt.Sprintf("resourceType eq '%s'", resourceType)
+	resources, err := a.resources.ListByResourceGroup(ctx, clusterRGName, filter, "", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	armResources := make([]arm.Resource, 0)
+	for _, res := range resources {
+		if *res.Type == "Microsoft.Compute/virtualMachines" {
+			vm, err := a.virtualMachines.Get(ctx, clusterRGName, *res.Name, mgmtcompute.InstanceView)
+			if err != nil {
+				a.log.Warn(err)
+				armResources = append(armResources, arm.Resource{
+					Resource: res,
+				})
+				continue
+			}
+			armResources = append(armResources, arm.Resource{
+				Resource: vm,
+			})
+		}
```

Comment:
I'm not sure we need this block since this endpoint would be to give an overview of the resources requested. If we do want to continue to get extra information just for `virtualMachines` we should wrap lines `42-44` in an else block to this condition. Otherwise we would be adding the resource info twice for `virtualMachines`